### PR TITLE
HARP-5043: Add missing techniques on omv-datasource

### PR DESCRIPTION
**Pull request by pit-rpg**
Signed-off-by: Stepan <11714928+pit-rpg@users.noreply.github.com>

### CODE REVIEW

**Comment by zbigg** [comment]
How that differs from closed #343 ?

Please describe look at comments to closed PR, because most important ones applies here too:
1) describe actual changes (IOW, missing is not enough)
2) try not to reformat code, example in comment to #343

File: @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts [hunk 1]
```diff
@@ -355,28 +304,37 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
             const techniqueName = technique.name;
 
             if (techniqueName === "line") {
-                applyLineTechnique(
+                this.applyLineTechnique(
```
Comment (amsgames94) [inline @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts]: This triple if statement could be easily modified to be a single one where the inputs can be computed by simple ternary operators (``const d = a ? b: c;``). Would add some readibility to the code.

File: @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts [hunk 2]
```diff
@@ -562,89 +520,258 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
...
                 }
-                rings.push(new Ring(ringContour, ringEdges, ringTexCoords));
+                polygons.push(rings);
             }
-            polygons.push(rings);
+
+            if (technique.name === "fill" || technique.name === "extruded-polygon") {
+                this.applyPolygonTechnique(polygons, technique, techniqueIndex, featureId, env);
+            } else if (technique.name === "line") {
+                polygons.forEach(rings => {
```
Comment (amsgames94) [inline @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts]: Same as above. A lot of duplicated code that breaks readability. Would be nice to simplify.

File: @here/harp-omv-datasource/lib/OmvDecoder.ts [hunk 3]
```diff
@@ -80,6 +80,43 @@ export class Ring {
 
         return area / 2;
     }
+
+    getOutlineLines() {
```
Comment (amsgames94) [inline @here/harp-omv-datasource/lib/OmvDecoder.ts]: ``getOutlineLines`` sounds a bit funny to me...
```suggestion
    getOutlines() {
```

Comment (amsgames94) [reply @here/harp-omv-datasource/lib/OmvDecoder.ts]: Also, cannot really say by looking at this function if it behaves as it should. Could you please add some unit tests for this?

File: @here/harp-omv-datasource/lib/VTJsonDataAdapter.ts [hunk 4]
```diff
@@ -150,7 +150,7 @@ export class VTJsonDataAdapter implements OmvDataAdapter {
                         let maxY = 0;
 
                         const ring: IRing = { coordinates: [], outlines: [] };
-                        for (let coordIdx = 0; coordIdx < outline.length; ++coordIdx) {
+                        for (let coordIdx = 0; coordIdx < outline.length - 1; ++coordIdx) {
```
Comment (amsgames94) [inline @here/harp-omv-datasource/lib/VTJsonDataAdapter.ts]: Please revert this if you have no serious reasoning behind it.

**Comment by zbigg** [comment]
Just a friendly reminder to look at our hacking rules (https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md),

- please decribe this change properly in commit message (already discussed in other PR)
- please make it "one commit" (rebase -i with squash)

It's much easier to review when we comply to these standards

**Comment by SahinHERE** [comment]
BTW not sure how to communicate this proper, but this change introduced a visual regression for the berlin based themes. Here a screen shot of #370

which got in before #359 and a screen shot after #359 was merged.



Can you please take a look at this problem ?

**Pull request merged**